packages/pluggableWidgets/color-picker-web/src/components/ColorPicker.tsx

```diff
@@ -0,0 +1,152 @@
+import { ModeEnum, TypeEnum, DefaultColorsType, FormatEnum } from "../../typings/ColorPickerProps";
+import {
+    ColorState,
+    SketchPickerProps,
+    ChromePickerProps,
+    TwitterPickerProps,
+    HuePickerProps,
+    BlockPickerProps,
+    GithubPickerProps,
+    CirclePickerProps,
+    SwatchesPickerProps,
+    SliderPickerProps,
+    MaterialPickerProps,
+    CompactPickerProps
+} from "react-color";
+import classNames from "classnames";
+import { getColorPicker, parseColor, validateColorFormat } from "../utils";
+import { Input } from "./Input";
+import { Button } from "./Button";
+
+export interface ColorPickerProps {
+    id: string;
+    name: string;
+    tabIndex?: number | undefined;
+    onChange: () => void;
+    onChangeComplete: (value: string) => void;
+    mode: ModeEnum;
+    type: TypeEnum;
+    color: string;
+    defaultColors: DefaultColorsType[];
+    format: FormatEnum;
+    disabled: boolean;
+    invalidFormatMessage?: string;
+}
+
+export type ColorPickerConfigProps =
+    | SketchPickerProps
+    | ChromePickerProps
+    | TwitterPickerProps
+    | HuePickerProps
+    | BlockPickerProps
+    | GithubPickerProps
+    | CirclePickerProps
+    | SwatchesPickerProps
+    | SliderPickerProps
+    | MaterialPickerProps
+    | CompactPickerProps;
+export const ColorPicker = (props: ColorPickerProps): ReactElement => {
+    const defaultColor = {
+        hex: "#FFFFFF",
+        rgb: "rgb(255,255,255)",
+        rgba: "rgb(255,255,255,1)"
+    };
+    const { type, mode, disabled, defaultColors, color, format, invalidFormatMessage } = props;
+    const colorElement = getColorPicker(type);
+    const [hidden, setHidden] = useState(mode !== "inline");
+    const [config, setConfig] = useState<ColorPickerConfigProps>({});
+    const [currentColor, setCurrentColor] = useState<string>(color || defaultColor[format]);
+    const [alertMessage, setAlertMessage] = useState<string | undefined>();
+
+    const validateColor = (color: string): void => {
+        const validFormat = validateColorFormat(color, format);
+        const alertMessage = validFormat ? invalidFormatMessage?.replace(/\{1}/, validFormat) : undefined;
+        setAlertMessage(alertMessage);
+    };
+    const onToggle = useCallback(
+        (hide: boolean): void => {
+            if (!disabled && mode !== "inline") {
+                setHidden(hide);
+            }
+        },
+        [disabled, mode]
+    );
+    const onColorValueUpdate = useCallback(
+        (color: ColorState): void => {
+            if (color && !disabled) {
+                const finalColor = parseColor(color, format);
+                setCurrentColor(finalColor);
+                props.onChangeComplete(finalColor);
+            }
+        },
+        [disabled, format, props]
+    );
+
+    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        const value = event.target.value;
+        validateColor(color);
```

Comment:
- This checks the `color` from the props, isn't that the old color? Shouldn't we check the new color?
- If the color is not valid, we still change the current color and do the `onChangeComplete`. Is that expected?

Reply:
- First point, yes, that is a bug.
- Second point, I just kept functionality, it was working like this before, it's just a warning message when format is incorrect, but data could be submitted, so I guess the question now do we want to change that or not? @diego-antonelli

Reply:
Prob good to direct it to @cjfhodges

Reply:
Is this the format of the color the user inputs to check whether its in the RGB, Hex or CMYK format for example

Reply:
Yes, but the problem is if the value is not valid it still could be submitted as a color value. The previous integration allowed to submit value and I kept that. So the question now should we submit incorrect color value or not?